gcs-action/src/gcs.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  Storage,
+  UploadResponse,
+  DeleteFileResponse,
+  File
+} from '@google-cloud/storage'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as core from '@actions/core'
+
+const storage = new Storage()
+
+/**
+ * Uploads a file to a bucket
+ * based on https://github.com/googleapis/nodejs-storage/blob/master/samples/uploadFile.js
+ * @param bucketName The name of the bucket
+ * @param filename The file path
+ * @param destination The destination prefix
+ */
+export async function uploadFile(
+  bucketName: string,
+  filename: string,
+  destination?: string
+): Promise<UploadResponse | Error> {
+  interface UploadOptions {
+    gzip: boolean
+    destination?: string
+  }
+  const options: UploadOptions = {gzip: true}
+  if (destination) {
+    options.destination = `${destination}${filename.slice(
```

Comment:
It'd be helpful to have a comment about why this mutation is necessary.

Reply:
simplified this to just use destination/name.